pkg/windows/eventwatcher/power_event_watcher_windows.go

```diff
@@ -0,0 +1,162 @@
+//go:build windows
+// +build windows
+
+package eventwatcher
+
+import (
+	"encoding/xml"
+	"fmt"
+	"syscall"
+	"unsafe"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"golang.org/x/text/encoding/unicode"
+)
+
+type (
+	eventLogEntry struct {
+		XMLName xml.Name `xml:"Event"`
+		System  System   `xml:"System"`
+	}
+
+	System struct {
+		EventID int `xml:"EventID"`
+	}
+
+	powerEventWatcher struct {
+		logger                  log.Logger
+		subscriptionHandle      uintptr
+		subscribeProcedure      *syscall.LazyProc
+		unsubscribeProcedure    *syscall.LazyProc
+		renderEventLogProcedure *syscall.LazyProc
+	}
+)
+
+const (
+	eventIdEnteringModernStandby = 506
+	eventIdExitingModernStandby  = 507
+	eventIdEnteringSleep         = 42
+
+	operationSuccessfulMsg = "The operation completed successfully."
+)
+
+func New(logger log.Logger) *powerEventWatcher {
+	evtApi := syscall.NewLazyDLL("wevtapi.dll")
+
+	return &powerEventWatcher{
+		logger:                  logger,
+		subscribeProcedure:      evtApi.NewProc("EvtSubscribe"),
+		unsubscribeProcedure:    evtApi.NewProc("EvtClose"),
+		renderEventLogProcedure: evtApi.NewProc("EvtRender"),
+	}
+}
+
+// subscribeToPowerEvents sets up a subscription to relevant power events with a callback to `onPowerEvent`.
+func (p *powerEventWatcher) SubscribeToPowerEvents() {
+	// WINEVENT_CHANNEL_GLOBAL_SYSTEM is "System"
+	channelPath, err := syscall.UTF16PtrFromString("System")
+	if err != nil {
+		level.Debug(p.logger).Log("msg", "error creating pointer to channel path", "err", err)
+		return
+	}
```

Comment:
Hrm. I wonder if we should log and return, or return the error. Feels a little TBD. 

This feels a bit like our usual go actors -- and we should return the error. Though we might end up ignoring it at the caller level

Reply:
I was also thinking this should maybe just be a rungroup actor, and that's why I was waffling on the error handling here -- because I didn't want to return an error on the equivalent of `Execute` and force a launcher restart, since this is just for logging right now.

I think the option that feels the most consistent to me is to move this setup work into the `New` function, and return an error on failure to set up. If `New` returns an error, log it and move on; if it doesn't, add the rungroup with a no-op `Execute`. Does that sound reasonable?

Reply:
Pushed this change, lmk what you think!

Reply:
Yes. I think that pattern is good.

`New` returns an error, and the caller decides if it's fatal. And after that, it's in it's own little world.